### PR TITLE
Reset all battler IDs at battle start

### PR DESCRIPTION
**Pull request by ghoulslash**
I found an edge case in my game where after a double/sos battle, gBattleScripting.battler might hold a battler index of say, 3. When you start the next battle with switch in weather or terrain, the `playanimation` command takes BS_SCRIPTING as an arg and marks that battler for a controller execution, but that ID can be invalid from a previous double battle. So this PR resets a few other battler ID global variables in `BattleStartClearSetData` and just to be extra explicit, resets them in `ABILITYEFFECT_SWITCH_IN_WEATHER` and `ABILITYEFFECT_SWITCH_IN_TERRAIN` as well.

### CODE REVIEW

File: src/battle_main.c [hunk 1]
```diff
@@ -3110,6 +3110,9 @@ static void BattleStartClearSetData(void)
 
     gBattlerAttacker = 0;
     gBattlerTarget = 0;
+    gEffectBattler = 0;
```
Comment (DizzyEggg) [inline src/battle_main.c:3113]: This makes sense.

File: src/battle_util.c [hunk 2]
```diff
@@ -4179,6 +4179,7 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
                 break;
             }
 
+            gBattleScripting.battler = 0;
```
Comment (DizzyEggg) [inline src/battle_util.c:4182]: Why is this needed? gBattleScripting.battler was set at the beginning of the switch's case to the value of battler.

File: src/battle_util.c [hunk 3]
```diff
@@ -4230,6 +4232,7 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
         if (effect != 0)
         {
             gBattleCommunication[MULTISTRING_CHOOSER] = GetCurrentWeather();
+            gBattleScripting.battler = 0;
```
Comment (DizzyEggg) [inline src/battle_util.c:4235]: Same as above.

File: src/battle_util.c [hunk 4]
```diff
@@ -4188,6 +4189,7 @@ u8 AbilityBattleEffects(u8 caseID, u8 battler, u16 ability, u8 special, u16 move
             // overworld weather started rain, so just do electric terrain anim
             gFieldStatuses = (STATUS_FIELD_ELECTRIC_TERRAIN | STATUS_FIELD_TERRAIN_PERMANENT);
             gBattleCommunication[MULTISTRING_CHOOSER] = 2;
+            gBattleScripting.battler = 0;
```
Comment (DizzyEggg) [inline src/battle_util.c:4192]: Same as above.

**Review by DizzyEggg — CHANGES_REQUESTED** [review]

File: src/battle_util.c (hunk 2 above)
Comment (ghoulslash) [reply src/battle_util.c:4182]: Technically there's a scenario where a user adds another battle script activating before weather, or an item/ability effect that triggers on weather/terrain that iterates over battlescripting.battler

**Comment by ghoulslash** [comment]
`Sorry I only get back to you now. Could you provide exact steps to reproduce the issue you're describing?`
There's no scenario in vanilla, but let's say some user adds a weather seed item and uses gBattleScripting.battler as a loop index, or sets it to some battler Id in their code or battle script for the effect. ABILITY_EFFECT_ON_TERRAIN_SWITCH_IN will now pass an invalid gBattleScripting.battler index through playanimation again.

**Comment by DizzyEggg** [comment]
> `Sorry I only get back to you now. Could you provide exact steps to reproduce the issue you're describing?` There's no scenario in vanilla, but let's say some user adds a weather seed item and uses gBattleScripting.battler as a loop index, or sets it to some battler Id in their code or battle script for the effect. ABILITY_EFFECT_ON_TERRAIN_SWITCH_IN will now pass an invalid gBattleScripting.battler index through playanimation again.
> 
> It is, admittedly, a pretty far-fetched edge case, but I don't think redundancy is ever bad with these global battler index variables

Alright, thanks, I get it now. I've got 2 requests then:
1. Move gBattleScripting.battler = 0; to the battlescript BattleScript_OverworldTerrain, as to not duplicate the code.
2. BattleScript_OverworldWeatherStarts doesn't seem to use battle scripting's battler, so not sure if it's needed there?

**Comment by DizzyEggg** [comment]
Actually, another idea, wouldn't it suffice to just change 
BattleScript_OverworldTerrain::
	printfromtable gTerrainStringIds
	waitmessage B_WAIT_TIME_LONG
	playanimation BS_SCRIPTING, B_ANIM_RESTORE_BG
	call BattleScript_ActivateTerrainEffects
	end3

to

playanimation BS_BATTLER_0, B_ANIM_RESTORE_BG

**Comment by ghoulslash** [comment]
> playanimation BS_BATTLER_0, B_ANIM_RESTORE_BG

good suggestion. Just implemented this and removed the gBattleScripting.battler = 0 setting before each battle script call. Ready for re-review

**Review by DizzyEggg — APPROVED** [review]
